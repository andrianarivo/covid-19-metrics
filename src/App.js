import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Layout from './components/Layout';
import Home from './routes/Home';
import Details from './routes/Details';
import ElementDetail from './routes/ElementDetail';

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route index element={<Home />} />
          <Route path="details" element={<Details />}>
            <Route path=":slug" element={<ElementDetail />} />
          </Route>
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
