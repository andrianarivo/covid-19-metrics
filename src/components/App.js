import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import Layout from './Layout';
import Home from '../routes/Home';
import Details from '../routes/Details';
import Countries from '../routes/Countries';
import store from '../redux/store';

function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Layout />}>
            <Route index element={<Home />} />
            <Route path="details" element={<Details />}>
              <Route path=":slug" element={<Countries />} />
            </Route>
          </Route>
        </Routes>
      </BrowserRouter>
    </Provider>
  );
}

export default App;
