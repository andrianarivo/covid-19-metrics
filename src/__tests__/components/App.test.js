import { act, create } from 'react-test-renderer';
import App from '../../components/App';

describe('Testing App component', () => {
  test('renders correctly', async () => {
    let tree;
    await act(() => {
      tree = create(<App />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
