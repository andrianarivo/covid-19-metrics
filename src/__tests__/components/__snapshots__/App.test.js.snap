// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component renders correctly 1`] = `
Array [
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          data-testid="home-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
    </ul>
  </nav>,
  <div>
    This is the home page
    <ul>
      <li>
        <p>
          North America
        </p>
        <p>
          2389357
           
          active cases
        </p>
        <p>
          Wed Jun 28 2023
        </p>
        <button
          data-testid="button-North America"
          onClick={[Function]}
          type="button"
        >
          North America
        </button>
      </li>
      <li>
        <p>
          Asia
        </p>
        <p>
          14773130
           
          active cases
        </p>
        <p>
          Wed Jun 28 2023
        </p>
        <button
          data-testid="button-Asia"
          onClick={[Function]}
          type="button"
        >
          Asia
        </button>
      </li>
    </ul>
  </div>,
]
`;
